scripts/bump_major_version.py

```diff
@@ -0,0 +1,318 @@
+#
+# - Copy includes to a new `_includes/vXY.Z` folder
+#
+# - Copy images to a new versioned folder, and edit image link text in the
+# referencing documents as needed to point to the right places
+#
+# - Update the YAML config files with the new version info
+#
+# - Create a new sidebar JSON file
+#
+# - In the newly created XY.Z folder, remove the various "new in vXY.Z-1"
+# callouts from the copied files.
+#
+# Note: this list may not be exhaustive, see the code below for details.
+#
+# This script attempts to automate these tasks as much as possible.
+#
+# For example, to do all of the admin tasks around bumping the version of
+# CockroachDB from 20.1 to 20.2, and then surfacing the places where additional
+# edits may be needed, run:
+#
+# $ ./bump_major_version.py -d /path/to/docs -f 20.1 -t 20.2
+#
+# Note that running this command will generate a lot of output, since
+# it attempts to grep for the "old-versionish" things in the docs that
+# have been copied over into the new version.  These potential
+# references to the previous version will need to be manually reviewed
+# and edited by a human.
+#
+# Finally, note that this script is *only* for use when CockroachDB
+# does a major version release, since that is the only time we would
+# ever create a new version folder and copy files into it.  That means
+# this script would be used for a 20.1 -> 20.2 transition, but *not*
+# for a 20.1.1 to 20.1.2 transition.
+
+parser = argparse.ArgumentParser(
+    description="do admin tasks related to version bumps"
+)
+
+parser.add_argument("-f", "--from-version",
+                    required=True,
+                    help="previous version of CockroachDB")
+
+parser.add_argument("-t", "--to-version",
+                    required=True,
+                    help="next version of CockroachDB")
+
+parser.add_argument("-d", "--docs-directory",
+                    default=os.environ.get("COCKROACH_DOCS_REPO"),
+                    help="directory where you store 'cockroachdb/docs'")
+
+args = parser.parse_args()
+
+old_version = args.from_version
+new_version = args.to_version
+docs_directory = args.docs_directory
+
+# Every file we touch that has content in it will also be stored here.  At the
+# end we will iterate over the lines of each file and do some simple-minded
+# transformations (mostly regexes to check and bump version numbers, and remove
+# "New in vXX.1" tags).
+files_to_check = {}
+
+
+def version_string_name(version):
+    # Int -> String
+    return 'v' + str(version)
+
+
+old_version_name = version_string_name(old_version)
+new_version_name = version_string_name(new_version)
+
+
+def copy_files(type=None):
+    # Int Int -> IO! State!
+
+    marktree_p = False
+
+    if type == 'images':
+        subdirectory = 'images'
+    elif type == 'includes':
+        subdirectory = '_includes'
+        marktree_p = True
+    # This option is provided for clarity, but is not really needed.
+    elif type == 'docs':
```

Comment:
Do we need to set `marktree_p` to true here as well?

Reply:
Yes, thanks!  Here and in the default `else` case.  Scary working without tests - Thanks!!!